SpriteKit/MendeluSpriteKit/Game/WelcomeScreen.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  WelcomeScreen.swift
+//  MendeluSpriteKit
+//
+//  Created by Raul Batista on 08.11.2022.
+//
+
+import Foundation
+import SpriteKit
+
+final class WelcomeScreen: SKScene {
+    // MARK: Properties
+    private var newGameButton: TouchNode?
+    
+    var newGameButtonTapped: (() -> Void)?
+}
+
+// MARK: Public API
+extension WelcomeScreen {
+    override func didMove(to view: SKView) {
```

Comment:
Wouldn't it be better to do this kind of setup in `sceneDidLoad()` function instead? In the documentation it even states that `This method is intended to be overridden in a subclass. It is the preferred location to peform custom setup after the scene has been initialized or decoded.`

Reply:
According to this https://stackoverflow.com/questions/53647094/init-vs-didmove-vs-scenedidload-in-spritekit it's preferable to setup the content in didMove 😅  so for now let's stick to it.

Reply:
I think `didMove` has almost similar description and in addition, we are supposed to add new `SKNodes` in that method 😄 To be honest, I would rather keep using `didMove`, otherwise, it would just confuse students 😄

Reply:
I see, makes sense 🙂👍